tensorflow/lite/micro/docs/optimized_kernel_implementations.md

```diff
@@ -0,0 +1,191 @@
+important.
+
+
+# High-Level Steps
+
+1. Pick a benchmark that you would like to measure the performance for.
+   * Existing benchmarks are in the [benchmarks directory](../benchmarks).
+   * If none of the existing benchmarks capture your use-case, then please create
+     a github issue or start a thread on micro@tensorflow.org to figure out how to
+     add in a new benchmark.
+   * If adding a publicly-available benchmark to the TFLM codebase is determined
+     to be infeasible, then a fall-back would be to have an internal benchmark
+     that can be used to document the benefits of adding in the optimizations via
+     PR descriptions.
+   * Adding optimized code without any associated benchmarks will need very
+     strong justification and will most likely not be permitted.
+
+1. Do the groundwork and architecture needed to be able to add in optimizations
+   for your target (more details in the [software architecture](#software-architecture)
+   section).
+
+1. Create one pull request for each optimized kernel with the PR description
+   clearly stating the commands that were used to measure the performance
+   improvement.
+
+   * This context is important even if the toolchain is proprietary and there are
+     currently a small number of users.
+      * See [this PR](https://github.com/tensorflow/tensorflow/pull/47098) as an example.
+      * At minimum the latency with and without the particular optimized kernel
+      should be documented. [Additional context](https://github.com/tensorflow/tensorflow/pull/46746)
+      may also be desirable.
+   * Here is some [general guidance](https://testing.googleblog.com/2017/09/code-health-providing-context-with.html)
+     on writing [good PR descriptions](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
+
+## Why Not Optimize the Portable Reference Kernels?
+
+We would like to explicitly point out (as have others) that the reference kernel
+implementations are not performant and there are plenty of opportunities to
+speed them up. This is by design and the reference kernels are meant to be a
+shared starting point to then be optimized in a target specific optimized kernel
+implementation.
+
+Two previous discussions on this topic are on
+[PR #42477](https://github.com/tensorflow/tensorflow/pull/42477) and
+[PR #45227](https://github.com/tensorflow/tensorflow/pull/45227)
+
+Our current point of view on this topic is that while optimizing shared
+reference code in a portable manner is attractive, we are making an explicit
+choice to not go down that path and instead rely on target-specific optimized
+implementations. The TFLM codebase has a growing list of optimized kernel
+implementations, and we are investing in making the process of adding new
+implementations smoother.
+
+# Software Architecture
+
+We recommend viewing the optimized kernel architecture as composed of the following three modules:
```

Comment:
"The optimized kernel architecture is composed of the following three modules:" Feels a bit clearer.

Reply:
done.